File: flutter_project/lib/dtos/gameInfo-dto.dart
import 'package:flutter_project/common/enum.dart';
import 'package:flutter_project/service/game-data-service.dart';

class GameInfo {
  String? gameId;
  GameMode gameMode;
  String? cardId;
  List<String> waitingPlayers;

  GameInfo({
    this.gameId,
    required this.gameMode,
    this.cardId,
    required this.waitingPlayers,
  });
}

File: common/dto/game/game-info.dto.ts
import { GameMode } from '@common/enums/game-play/game-mode';

export interface GameInfo {
    id: string;
    gameMode: GameMode;
    cardId?: string;
    waitingPlayers: string[];
    observerNbr: number;
    playersWithAccess?: string[];
}
